src/main/java/com/ericsson/gerrit/plugins/eiffel/events/generators/EiffelSourceChangeCreatedEventGenerator.java

```diff
@@ -49,6 +60,9 @@ public static EiffelSourceChangeCreatedEvent generate(PatchSetCreatedEvent patch
         final int insertions = patchSetAttribute.sizeInsertions;
         final int deletions = patchSetAttribute.sizeDeletions;
         final String changeId = patchSetCreatedEvent.changeKey.toString();
+        final Link previousVersionLink = createPreviousVersionLink(pluginDirectoryPath, projectName,
```

Comment:
That would apply with the rest of the data as well but it would become to messy in my opinion. That's why we have one section where we retrieve all data and one section where we populate it.